pwa/src/templates/templateParts/sourcesForm/SourcesAuthFormTemplate.tsx

```diff
@@ -24,59 +25,95 @@ export const SourcesAuthFormTemplate: React.FC<SourcesAuthFormTemplateProps & Re
   errors,
   disabled,
 }) => {
-  const { t } = useTranslation();
-
   switch (selectedAuth) {
-    case "none":
-      return <></>;
     case "apikey":
-      return (
-        <FormField>
-          <FormFieldInput>
-            <FormFieldLabel>{t("Api key")}</FormFieldLabel>
-            <Textarea {...{ register, errors, disabled }} name="apikey" validation={{ maxLength: 225 }} />
-            {errors["apikey"] && <ErrorMessage message={errors["apikey"].message} />}
-          </FormFieldInput>
-        </FormField>
-      );
+      return <ApiKeyForm {...{ register, errors, disabled }} />;
```

Comment:
Instead of passing all these props explicitly, we can spread them.

First, we need to not explicitly destructure them in our component, like so:

```
export const SourcesAuthFormTemplate: React.FC<SourcesAuthFormTemplateProps & ReactHookFormProps> = ({
  selectedAuth, <= we need this one _within_ this component, so we have to destructure it
  ...rest <= we're not actually using any of these, so there's not need to destruct
}) => {
```

Then, in our switch, we can call the components, like so:

```
return <ApiKeyForm {...rest} />;
```

You could do the same in the underlying components (ApiKeyForm, ...), however, that might be overkill.